engine/src/flutter/shell/platform/windows/flutter_window.cc

```diff
@@ -537,47 +541,71 @@ FlutterWindow::HandleMessage(UINT const message,
       break;
-    case WM_TOUCH: {
-      UINT num_points = LOWORD(wparam);
-      touch_points_.resize(num_points);
-      auto touch_input_handle = reinterpret_cast<HTOUCHINPUT>(lparam);
-      if (GetTouchInputInfo(touch_input_handle, num_points,
-                            touch_points_.data(), sizeof(TOUCHINPUT))) {
-        for (const auto& touch : touch_points_) {
-          // Generate a mapped ID for the Windows-provided touch ID
-          auto touch_id = touch_id_generator_.GetGeneratedId(touch.dwID);
-
-          POINT pt = {TOUCH_COORD_TO_PIXEL(touch.x),
-                      TOUCH_COORD_TO_PIXEL(touch.y)};
-          ScreenToClient(window_handle_, &pt);
-          auto x = static_cast<double>(pt.x);
-          auto y = static_cast<double>(pt.y);
-
-          if (touch.dwFlags & TOUCHEVENTF_DOWN) {
-            OnPointerDown(x, y, kFlutterPointerDeviceKindTouch, touch_id,
-                          WM_LBUTTONDOWN);
-          } else if (touch.dwFlags & TOUCHEVENTF_MOVE) {
-            OnPointerMove(x, y, kFlutterPointerDeviceKindTouch, touch_id, 0);
-          } else if (touch.dwFlags & TOUCHEVENTF_UP) {
-            OnPointerUp(x, y, kFlutterPointerDeviceKindTouch, touch_id,
-                        WM_LBUTTONDOWN);
-            OnPointerLeave(x, y, kFlutterPointerDeviceKindTouch, touch_id);
-            touch_id_generator_.ReleaseNumber(touch.dwID);
+    case WM_POINTERDOWN:
+    case WM_POINTERUPDATE:
+    case WM_POINTERUP:
+    case WM_POINTERLEAVE: {
+      x_pos = GET_X_LPARAM(lparam);
+      y_pos = GET_Y_LPARAM(lparam);
+      auto const x = static_cast<double>(x_pos);
+      auto const y = static_cast<double>(y_pos);
+      auto const pointerId = GET_POINTERID_WPARAM(wparam);
+      POINTER_INFO pointerInfo;
+      if (windows_proc_table_->GetPointerInfo(pointerId, &pointerInfo)) {
+        UINT32 pressure = 0;
+        UINT32 rotation = 0;
+        if (pointerInfo.pointerType == PT_PEN) {
+          POINTER_PEN_INFO penInfo;
+          if (windows_proc_table_->GetPointerPenInfo(pointerId, &penInfo)) {
+            pressure = penInfo.pressure;
+            rotation = penInfo.rotation;
           }
         }
-        CloseTouchInputHandle(touch_input_handle);
+        auto touch_id = touch_id_generator_.GetGeneratedId(pointerId);
+        FlutterPointerDeviceKind device_kind = kFlutterPointerDeviceKindMouse;
+        switch (pointerInfo.pointerType) {
+          case PT_TOUCH:
+            device_kind = kFlutterPointerDeviceKindTouch;
+            break;
+          case PT_PEN:
+            device_kind = kFlutterPointerDeviceKindStylus;
+            break;
+          case PT_MOUSE:
+            device_kind = kFlutterPointerDeviceKindMouse;
+            break;
+          case PT_TOUCHPAD:
+            device_kind = kFlutterPointerDeviceKindTrackpad;
+            break;
+          default:
+            assert(false, 'Unrecognized device key {}', pointerInfo.pointerType);
+            return kFlutterPlane.value;
```

Comment:
I don't think that we use `assert` in the code often? Could this be:

```suggestion
            FML_LOG(ERROR) << "Unrecognized device key " << pointerInfo.pointerType;
            break;
```

I would suggest `break`ing here just so that we don't "handle" a message that we don't know about. It can bubble down to Win32

Reply:
Sure, I'm not an expert in this codebase, I just looked how it is handled in other places. i commited the suggestion